Index: Modules/main.c
===================================================================
--- Modules/main.c
+++ Modules/main.c	2008-12-17 23:52:15.000000000 +0100
@@ -105,9 +105,9 @@
 	if (exitcode)
 		fprintf(f, "Try `python -h' for more information.\n");
 	else {
-		fprintf(f, usage_1);
-		fprintf(f, usage_2);
-		fprintf(f, usage_3);
+		fprintf(f, "%s", usage_1);
+		fprintf(f, "%s", usage_2);
+		fprintf(f, "%s", usage_3);
 		fprintf(f, usage_4, DELIM, DELIM, PYTHONHOMEHELP);
 	}
 #if defined(__VMS)
Index: Objects/unicodeobject.c
===================================================================
--- Objects/unicodeobject.c
+++ Objects/unicodeobject.c	2008-12-17 23:50:11.000000000 +0100
@@ -839,7 +839,7 @@
     if (restuple == NULL)
 	goto onError;
     if (!PyTuple_Check(restuple)) {
-	PyErr_Format(PyExc_TypeError, &argparse[4]);
+	PyErr_Format(PyExc_TypeError, "%s", &argparse[4]);
 	goto onError;
     }
     if (!PyArg_ParseTuple(restuple, argparse, &PyUnicode_Type, &repunicode, &newpos))
@@ -2564,7 +2564,7 @@
     if (restuple == NULL)
 	return NULL;
     if (!PyTuple_Check(restuple)) {
-	PyErr_Format(PyExc_TypeError, &argparse[4]);
+	PyErr_Format(PyExc_TypeError, "%s", &argparse[4]);
 	Py_DECREF(restuple);
 	return NULL;
     }
@@ -3837,7 +3837,7 @@
     if (restuple == NULL)
 	return NULL;
     if (!PyTuple_Check(restuple)) {
-	PyErr_Format(PyExc_TypeError, &argparse[4]);
+	PyErr_Format(PyExc_TypeError, "%s", &argparse[4]);
 	Py_DECREF(restuple);
 	return NULL;
     }
